frontend/app/components/DotMenu.tsx

```diff
@@ -0,0 +1,389 @@
+import { useState } from 'react';
+import { Text, TouchableOpacity, View, Share } from 'react-native';
+import { Ionicons } from '@expo/vector-icons';
+import { MotiView } from 'moti';
+import { CardData } from '../types/card';
+import { setStringAsync } from 'expo-clipboard';
+
+type IoniconsName = keyof typeof Ionicons.glyphMap;
+
+interface DotMenuProps {
+  item: CardData;
+  onStatsToggle: () => void;
+  CARD_WIDTH: number;
+  CARD_HEIGHT: number;
+}
+
+interface MenuOption {
+  id: string;
+  title: string;
+  icon: IoniconsName;
+  action: () => void;
+}
+
+const DotMenu: React.FC<DotMenuProps> = ({
+  item,
+  onStatsToggle,
+  CARD_WIDTH,
+  CARD_HEIGHT,
+}) => {
+  const [isMenuVisible, setIsMenuVisible] = useState(false);
+  const [showReportMenu, setShowReportMenu] = useState(false);
+  const [showNotification, setShowNotification] = useState(false);
+  const [notificationMessage, setNotificationMessage] = useState('');
+
+  const showSuccessNotification = (message: string) => {
+    setNotificationMessage(message);
+    setShowNotification(true);
+    setTimeout(() => setShowNotification(false), 2000);
+  };
+
+  const handleMenuPress = () => {
+    setIsMenuVisible(true);
+  };
+
+  const handleSharePress = async () => {
+    setIsMenuVisible(false);
+    try {
+      await Share.share({
+        message: `${item.text}\n\nMeaning: ${item.meaning}\n\nExample: ${item.examples[0] || 'No example available'}`,
+        title: 'Check out this idiom!',
+      });
```

Comment:
Nice :) in the future we should also add here a link to the app or something like htat